### PR TITLE
Task 5529 mailer registration

### PR DESCRIPTION
**Pull request by chuxuanthang**
https://edu-redmine.framgia.vn/issues/5529

### CODE REVIEW

File: app/Http/Controllers/LoginController.php [hunk 1]
```diff
@@ -32,9 +32,12 @@ public function postLogin(LoginRequest $request)
             return redirect()->route('home.index');
         }
         $input = $request->all();
-
         if (!empty($input)) {
             if (Auth::attempt(['email' => $input['email'], 'password' => $input['password']])) {
+                if (is_null(Auth::user()->email_verified_at)) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 37</br>Blank line found at start of control structure

Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 37</br>Blank line found at start of control structure

Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 37</br>Blank line found at start of control structure

Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 37</br>Blank line found at start of control structure

File: app/Http/Controllers/LoginController.php [hunk 2]
```diff
@@ -32,9 +34,12 @@ public function postLogin(LoginRequest $request)
...
```
Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 39</br>Blank line found at start of control structure

File: app/Http/Controllers/LoginController.php [hunk 3]
```diff
@@ -59,18 +64,32 @@ public function signup()
...
+        $user = $this->userRepository->find($id);
+        if ($user->hasVerifiedEmail()) {
+            return redirect()->route('home.index');
+        }
+        event(new Registered($user));
+        \Session::flash('resent', __('alert.success.resendSuccess'));
+
+        return view('home.account.verify', compact('user'));
+    }
+}
```
Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Files.EndFileNewline.NoneFound</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 95</br>Expected 1 newline at end of file; 0 found

File: app/Http/Controllers/LoginController.php (hunk 2 above)
Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 39</br>Blank line found at start of control structure

File: app/Http/Controllers/LoginController.php (hunk 3 above)
Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Files.EndFileNewline.NoneFound</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 95</br>Expected 1 newline at end of file; 0 found

File: app/Http/Controllers/LoginController.php (hunk 2 above)
Comment (framgiaciv3) [inline app/Http/Controllers/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/LoginController.php</br>**Line**: 39</br>Blank line found at start of control structure

File: app/Providers/AppServiceProvider.php [hunk 4]
```diff
@@ -15,6 +18,22 @@ class AppServiceProvider extends ServiceProvider
...
+
+        VerifyEmail::toMailUsing(function ($notifiable) {
+            $verifyUrl = URL::temporarySignedRoute(
+                'home.email.verify',
+                now()->addMinutes(60),
+                ['id' => $notifiable->getKey()]
+            );
+
+            return (new MailMessage())
+                ->subject( __('email.subject'))
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.SpaceAfterOpenBracket</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 30</br>Space after opening parenthesis of function call prohibited

File: app/Providers/AppServiceProvider.php [hunk 5]
```diff
@@ -15,6 +18,22 @@ class AppServiceProvider extends ServiceProvider
...
+                ->greeting( __('email.greeting'))
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.SpaceAfterOpenBracket</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 31</br>Space after opening parenthesis of function call prohibited

File: app/Providers/AppServiceProvider.php (hunk 4 above)
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.SpaceAfterOpenBracket</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 30</br>Space after opening parenthesis of function call prohibited

File: app/Providers/AppServiceProvider.php (hunk 5 above)
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.SpaceAfterOpenBracket</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 31</br>Space after opening parenthesis of function call prohibited

**Comment by chuxuanthang** [comment]
ready

File: app/Http/Controllers/LoginController.php [hunk 6]
```diff
@@ -59,18 +65,47 @@ public function signup()
...
-
         if (!empty($input)) {
             $input['password'] = bcrypt($input['password']);
             $input['fullname'] = 'your fullname';
-
-            if ($this->userRepository->create($input)) {
-                \Session::flash('success_alert', __('alert.success.signup'));
-
-                return redirect()->route('home.index');
+            $input['gender'] = '1';
```
Comment (nguyenngocsond) [inline app/Http/Controllers/LoginController.php]: đưa vào config mấy biến 1 2 3 kiểu này nhé

File: app/Http/Controllers/LoginController.php [hunk 7]
```diff
@@ -32,9 +35,12 @@ public function postLogin(LoginRequest $request)
             return redirect()->route('home.index');
         }
         $input = $request->all();
-
         if (!empty($input)) {
             if (Auth::attempt(['email' => $input['email'], 'password' => $input['password']])) {
+                if (is_null(Auth::user()->email_verified_at)) {
+                    Auth::logout();
+                    return redirect()->route('home.login')->withErrors(['incorrectInfo' => __('alert.error.login')]);
```
Comment (danhloimta) [inline app/Http/Controllers/LoginController.php:43]: thiếu dòng mới trc return

File: public/css/app.css [hunk 8]
```diff
@@ -10490,6 +10490,234 @@ a.text-dark:focus {
   }
 }
 
+.toast-title {
+  font-weight: bold;
```
Comment (BunnyPi04) [inline public/css/app.css]: tab lại cả file
sao lại up css trong public lên vậy?

File: app/Http/Controllers/LoginController.php [hunk 9]
```diff
@@ -59,18 +66,47 @@ public function signup()
...
+            $input['fullname'] = config('constants.DEFAULT_USER_FULLNAME');
+            $input['gender'] = config('constants.DEFAULT_USER_GENDER');
+            $input['image_path'] = config('constants.IMAGE_UPLOAD_PATH') . config('constants.DEFAULT_USER_IMAGE');
+            $input['role_id'] = config('constants.DEFAULT_USER_ROLE_ID');
+            $input['status'] = config('constants.DEFAULT_USER_STATUS');
+            if ($user = $this->userRepository->create($input)) {
+                event(new Registered($user));
+
+                return view('home.account.verify', compact('user'));
             } else {
```
Comment (thanhpv958) [inline app/Http/Controllers/LoginController.php]: if return rồi thì ko cần else.

**Comment by chuxuanthang** [comment]
ready

**Pull request merged**